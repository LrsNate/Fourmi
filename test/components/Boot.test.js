import { render, shallow } from "enzyme";
import React from "react";

import {
  Boot,
  mapDispatchToProps,
  mapStateToProps
} from "../../app/components/Boot";

jest.mock("../../app/actions/database", () => ({
  ensureDatabaseFolderExistsAction: jest.fn(() =>
    Promise.resolve("ensureDatabaseFolderExists")
  ),
  ensureDatabaseExistsAction: jest.fn(() =>
    Promise.resolve("ensureDatabaseExists")
  )
}));

jest.mock("../../app/lib/files", () => ({
  getDataFolderPath: jest.fn(() => "/home/foo/stuff")
}));

describe("The boot component", () => {
  const messages = [
    "DATABASE_FOLDER_NOT_FOUND",
    "DATABASE_FOLDER_FOUND",
    "DATABASE_FOLDER_READY",
    "DATABASE_NOT_FOUND",
    "DATABASE_FOUND",
    "DATABASE_READY"
  ];

  it("matches the snapshot", () => {
    const rendered = render(
      <Boot messages={messages} initializeDatabase={jest.fn()} />
    );

    expect(rendered).toMatchSnapshot();
  });

  it("triggers a database initialization", () => {
    const initializeDatabase = jest.fn();

    shallow(
      <Boot messages={messages} initializeDatabase={initializeDatabase} />
    );

    expect(initializeDatabase).toHaveBeenCalled();
  });
});

describe("The mapStateToProps function", () => {
  it("transmits status history", () => {
    const state = { database: { statusHistory: ["foo", "bar"] } };

    const { messages } = mapStateToProps(state);

    expect(messages).toEqual(["foo", "bar"]);
  });
});

describe("The mapDispatchToProps function", () => {
  it("dispatches the database folder and file checks", () => {
    const dispatch = jest.fn(i => i);

    mapDispatchToProps(dispatch)
      .initializeDatabase()
      .then(dispatched => {
        expect(dispatch).toHaveBeenCalledTimes(2);
        expect(dispatched).toBe("ensureDatabaseExists");
      });
  });
});
