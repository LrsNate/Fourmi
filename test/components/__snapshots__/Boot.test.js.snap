// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The boot component matches the snapshot 1`] = `
<ul>
  <li>
    Le dossier de données n'a pas pu être trouvé ; création du dossier "/home/foo/stuff"...
  </li>
  <li>
    Le dossier de données existe à : "/home/foo/stuff".
  </li>
  <li>
    Le dossier de données est prêt.
  </li>
  <li>
    La base de données n'a pas pu être trouvée. Téléchargement en cours...
  </li>
  <li>
    La base de données a été trouvée.
  </li>
  <li>
    La base de données est prête. Redirection en cours...
  </li>
</ul>
`;
